services/common/boltzmann/README.md

```diff
@@ -0,0 +1,137 @@
+
+Here's the simplest possible handler:
+
+```js
+const { Response } = require('node-fetch');
+
+async function greeting() {
+  return new Response('hello world!');
+}
+```
+
+We have wrapped up the somewhat awkward Response api in some conveniences that handle the most common response types we've had to use. Let's respond with json:
+
+```js
+const { response } = require('boltzmann');
+
+async function greeting() {
+  return response.json({ greeting: 'hello world!' });
+}
+```
+
+Let's build a router and define one parametrized route on it:
+
+```js
+const { response, fork } = require('boltzmann');
+
+function makeRouter() {
+  const router = fork.router()(
+    fork.get('/greet/:human', greeting)
+  );
+
+  return router;
+}
+
+async function greeting(context, params) {
+  return response.text(`greetings, ${params.human}`);
+}
+```
+
+Here are all the response conveniences:
+
+* `text(body, status = 200, extraHeaders = {})`: Respond with plain text.
+* `json(body, status = 200, extraHeaders = {})`: Respond with a json object.
+* `message(msg, status = 200, extraHeaders = {})`: Wrap the given text message with the object expected by the ds cli.
+* `bytes(stream, status = 200, extraHeaders = {})`: Set up an octet byte stream response.
+* `html(text, status = 200, extraHeaders = {}) `: Set up an html response.
+* `redirect(where, status = 301, extraHeaders = {})`: Redirect the request. Note that this breaks with the pattern of the other functions: you pass the url to redirect to, not a response body.
+* `authneeded(message, status = 401, extraHeaders = {})`: Respond with www-authenticate and the given message.
+* `error(err, status = 500, extraHeaders = {})`: Respond with the given error. You can pass a string if you want, but an error object is preferred.
+
+## Middleware functions
+
+
+
+Here's Boltmann's redis middleware:
+
+```js
+// build the middleware function, using runtime config
+function middlewareBuilder({
+  redisURL = process.env.REDIS_URL || 'redis://localhost:6379'
+} = {}) {
+  // this function gets passed in the middleware list
+  return function theMiddlewareFunction(next) {
+    const client = redis.createClient(redisURL);
+
+	// The innermost function is what gets run on every request.
+    return function thisIsRunOnEveryRequest(context) {
+      context.redis = client;
+      return next(context);
+    };
+  };
+}
+```
+
+## Putting it all together...
+
+
+Here's the Boltzmann hello world, complete:
+
+
+```js
+'use strict';
+
+const boltzmann = require('./index.js');
+
+async function greeting(context, params) {
+  return boltzmann.response.text(`greetings, ${params.human}`);
+}
+
+const router = boltzmann.fork.router()(
+    fork.get('/greet/:human', greeting)
+);
+
+const myMiddles = [
+	boltmmann.middleware['logger'],
```

Comment:
ESLint might complain that this should be written `boltzmann.middleware.logger`. That said, I'd also be okay with folks just doing `require('boltzman/middleware/logger')` & not explicitly exporting the middleware through `middleware/index.js` (Saves us a step to remember in the future, at least.)